cpp/include/raft/linalg/axpy.cuh

```diff
@@ -62,66 +62,61 @@ void axpy(const raft::handle_t& handle,
  * @brief axpy function
  *  It computes the following equation: y = alpha * x + y
  *
- * @tparam InType  Type raft::device_mdspan
- * @tparam ScalarIdxType Index Type of scalar
  * @param [in] handle raft::handle_t
  * @param [in] alpha raft::device_scalar_view
  * @param [in] x Input vector
  * @param [inout] y Output vector
  */
-template <typename InType,
-          typename OutType,
-          typename ScalarIdxType,
-          typename = raft::enable_if_input_device_mdspan<InType>,
-          typename = raft::enable_if_output_device_mdspan<OutType>>
+template <typename ElementType,
+          typename IndexType       = std::uint32_t,
```

Comment:
We should consider allowing the compiler to infer these things. In general we have been removing the defaults everywhere unless absolutely necessary because they create confusion when a user actually wants to use a type other than the default (the compiler complains a valid overload doesn't exist unless the user explicitly provides the template arg)